cg/meta/workflow/nf_analysis.py

```diff
@@ -803,3 +805,34 @@ def get_cases_to_store(self) -> list[Case]:
             if self.trailblazer_api.is_latest_analysis_completed(case_id=case.internal_id)
             or self.trailblazer_api.is_latest_analysis_qc(case_id=case.internal_id)
         ]
+
+    def clean_workflow_run_dir(self, yes: bool, case_id: str, dry_run: bool = False):
+        """Remove workflow run directory."""
+
+        self.check_analysis_ongoing(case_id=case_id)
+        analysis_path: list[Path] | Path = self.get_case_path(case_id)
+
+        if dry_run:
+            LOG.info(f"Would have deleted: {analysis_path}")
+            return EXIT_SUCCESS
+
+        self.clean_run_dir(case_id=case_id, yes=yes, case_path=analysis_path)
+
+    def clean_past_run_dirs(self, before_str: str, yes: bool = False, dry_run: bool = False):
```

Comment:
This method seems almost the same as `past_run_dirs` , can it be move to the more general analysis_api and refactor so we don't have duplicated methods?